pkg/operator/k8sutil/node.go

```diff
@@ -279,3 +281,62 @@ func NodeIsInRookNodeList(targetNodeName string, rookNodes []rookalpha.Node) boo
 	}
 	return false
 }
+
+// AddNodeAffinity will return v1.NodeAffinity or error
+func AddNodeAffinity(nodeAffinity string) (*v1.NodeAffinity, error) {
+	newNodeAffinity := &v1.NodeAffinity{
+		RequiredDuringSchedulingIgnoredDuringExecution: &v1.NodeSelector{
+			NodeSelectorTerms: []v1.NodeSelectorTerm{
+				{},
+			},
+		},
+	}
+	nodeLabels := strings.Split(nodeAffinity, ";")
```

Comment:
In your examples, e.g., `role=storage-node; storage=rook,ceph`, there is a space after the `;`, is that trimmed away? In a test https://play.golang.org/p/QZw_ZQZVFJk the spaces remained. Could that cause any issues?

Reply:
Nice catch. It should not cause any issue but i'll trim it away, just in case.